Docker

nikitabel0 opened the pull request:
- Upgrade PostgreSQL from 13 to 17
- Upgrade Python base image from 3.11 to 3.13
- Add environment variables configuration
- Add project-prefixed volume name

Commit:
add docker

Commit:
build: complete Docker stack upgrade

- Upgrade PostgreSQL from 13 to 17
- Upgrade Python base image from 3.11 to 3.13
- Add environment variables configuration
- Add project-prefixed volume name

Screxy commented on docker-compose.yml:
```diff
@@ -0,0 +1,28 @@
+    ports:
+      - "8000:8000"
+    environment:
+      - DATABASE_URL=postgresql+asyncpg://${POSTGRES_USER}:${POSTGRES_PASSWORD}@db:5432/${POSTGRES_DB}
```
nit: порт бы тоже вынести в переменную, да и вообще как будто урл к бд билдить на стороне прилы, а не в env сувать. но ладно

yakser commented on .gitignore:
```diff
@@ -63,8 +63,7 @@ Thumbs.db
 *.sublime-workspace
 
 # Docker
```
коммент надо убрать. потерялось из прошлого ревью

yakser commented on docker-compose.yml:
```diff
@@ -0,0 +1,28 @@
+version: '3.8'
```
надо убрать

yakser commented on docker-compose.yml:
```diff
@@ -0,0 +1,28 @@
+    ports:
+      - "5432:5432"
+    volumes:
+      - psychological_help_postgres_data:/var/lib/postgresql/data
```
psychological_help -> mospoly-psychological-pg-data

Commit:
refactor: migrate to environment-based configuration

- Remove hardcoded database URLs from LocalDevConfig and DevConfig
- Use single Config class that builds DATABASE_URL from environment variables
- Make RESET_DB_ON_START configurable via environment
- Simplify configuration structure
